File: Javascript/ImplementTrie.js
// Leetcode Problem - 208

var TrieNode = function(){
    this.children = new Map()
    this.word = false
}

var Trie = function(){
    this.root = new TrieNode()
}

Trie.prototype.insert = function(word){
    let cur = this.root

    for (const c of word){
        if (!cur.children.has(c)){
            cur.children.set(c,new TrieNode())
        }

        cur = cur.children.get(c)
    }

    cur.word = true
}
Trie.prototype.search = function(word){
    let cur = this.root

    for (const c of word){
        if (!cur.children.has(c)){
            return false
        }
        cur = cur.children.get(c)
    }

    return cur.word
}
Trie.prototype.startsWith = function(prefix){
    let cur = this.root

    for (const c of prefix){
        if (!cur.children.has(c)){
            return false
        }
        cur = cur.children.get(c)
    }

    return true
}

File: Python/ImplementTrie.py
## Leetcode Problem - 208

class TrieNode:
    def __init__(self) -> None:
        self.children = {}
        self.word = False


class Trie:

    def __init__(self) -> None:
        self.root = TrieNode()

    def insert(self,word):
        cur = self.root

        for c in word:
            if c not in cur.children:
                cur.children[c] = TrieNode()
            cur = cur.children[c]
        
        cur.word = True


    def search(self,word):
        cur = self.root

        for c in word:
            if c not in cur.children:
                return False
            cur = cur.children[c]
        

        return cur.word

    def startsWith(self,prefix):
        cur = self.root

        for c in prefix:
            if c not in cur.children:
                return False
            cur = cur.children[c]

        return True